### PR TITLE
映射文档 No. 27

### PR DESCRIPTION
**Pull request by zpc2024**
https://github.com/PaddlePaddle/docs/issues/5691
完成分组No.27文档映射，新提交10个md文件。 
paddle.utils.cpp_extension.load函数中出现interpreter 参数，源代码中没有这个参数，文档中没有添加。

### CODE REVIEW

File: docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## [ 参数不一致 ]torch.linalg.matrix_power
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md]: 分类应该为 `torch 参数更多`~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+### [paddle.linalg.matrix_power](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/linalg/matrix_power_cn.html)
+
+```python
+paddle.linalg.matrix_power(x,
+                        n,
+                        name=None)
+```
+
+两者功能一致但参数不一致，具体如下：
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md]: 这句话也需要修改为对应的

File: docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| A          |  x           | 输入的欲进行 n 次幂运算的一个或一批方阵，类型为 Tensor,仅参数名不一致。  |
+| n         | n         | 输入的幂次，类型为 int。 |
+|out         | -         |  表示输出的 Tensor ， Paddle 无此参数，需要进行转写。 |
+| -   |   name  |        可选，一般无需设置，默认值为 None，PyTorch 无此参数， Paddle 保持默认即可。                            |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md]: name参数映射可以直接删去

File: docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+### 转写示例
+#### out：指定输出
+```python
+# Pytorch 写法
+x = torch.tensor([[1, 2, 3], [4, 5, 6], [7,8,9]])
+torch.linalg.matrix_power(x, 3， out = y)
+
+# Paddle 写法
+x = paddle.to_tensor([[1, 2, 3], [4, 5, 6], [7,8,9]])
+paddle.assign(paddle.linalg.matrix_power(x, 3), y)
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md:34]: 转写还是直接写为y = paddle.linalg.matrix_power(x, 3)即可，同时为了示例的简便，可以不用定义`x = paddle.to_tensor([[1, 2, 3], [4, 5, 6], [7,8,9]])`， 直接使用即可，下面类似

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.dsplit.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+Paddle 相比 PyTorch 支持更多其他参数，具体如下：
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| input          |  x           | 输入多维 Tensor ，仅参数名不一致。  |
+| indices_or_sections         | num_or_sections         | int 或者仅含有 int 的 list 或者 tuple ，用于分割的参数，仅参数名不一致。 |
+| -         | axis         |     可选，默认为 0 ，表示需要分割的维度，PyTorch 无此参数，Paddle 保持默认即可。 |
+| -   |   name  |        可选，一般无需设置，默认值为 None。                                |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.dsplit.md]: name 参数映射部分都可以直接删去

File: docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.special.expm1.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+## [ 参数不一致 ]torch.special.expm1
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.special.expm1.md]: 这里的分类应该是 torch参数更多

File: docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.special.expm1.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+### [paddle.expm1](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/expm1_cn.html)
+
+```python
+paddle.expm1(x,
+        name=None)
+```
+
+两者功能一致但参数不一致，具体如下：
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.special.expm1.md]: 这句话也需要修改~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.cpp_extension.CUDAExtension.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+paddle.utils.cpp_extension.CUDAExtension(sources,
+                                    *args,
+                                    **kwargs)
+```
+
+Pytorch 相比 Paddle 支持更多其他参数，具体如下：
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| name          | -            | 参数 name，PaddlePaddle 无此参数。  |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.cpp_extension.CUDAExtension.md]: 可以补充上 `一般对网络训练结果影响不大，可直接删除。`

File: docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.cpp_extension.CUDAExtension.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+| sources         | sources         | 用于指定自定义 OP 对应的源码文件   |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.cpp_extension.CUDAExtension.md]: 末尾加一个句号

File: docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.cpp_extension.CppExtension.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+paddle.utils.cpp_extension.CppExtension(sources,
+                                    *args,
+                                    **kwargs)
+```
+
+Pytorch 相比 Paddle 支持更多其他参数，具体如下：
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| name          | -            | 参数 name，PaddlePaddle 无此参数。  |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.cpp_extension.CppExtension.md]: 可以补充上 `一般对网络训练结果影响不大，可直接删除。`

File: docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.data.random_split.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+paddle.io.random_split(dataset,
+                    lengths,
+                    generator=None)
+```
+
+两者参数和用法完全一致，具体如下：
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| dataset          | dataset            | 此参数必须是 paddle.io.Dataset 或 paddle.io.IterableDataset 的一个子类实例或实现了 __len__ 的 Python 对象，用于生成样本下标。默认值为 None。  |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.data.random_split.md]: 这里的备注不需要写必须是paddle.io.Dataset这部分，因为torch并没有这些类，直接写`表示可迭代数据集。` 即可

**Review by Fancy-hjyp — COMMENTED** [review]
* name 参数映射时可以去掉
* 转写示例时为了简洁，直接使用变量即可

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.take_along_dim.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
+## [ torch 参数更多 ]torch.take_along_dim
+### [torch.take_along_dim](https://pytorch.org/docs/1.13/generated/torch.take_along_dim.html#torch.take_along_dim)
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.take_along_dim.md]: paddle 也有对应api，但是官网没有更新，可以运行一下，他的参数就是paddle.take_along_axis()去掉arr参数，所以还是改为 torch.Tensor.take_along_dim 和 paddle.Tensor.take_along_axis。
pytorch参考链接为https://pytorch.org/docs/1.13/generated/torch.Tensor.take_along_dim.html?highlight=torch+tensor+take_along_dim#torch.Tensor.take_along_dim 
paddle参考链接为空即可
可以参考已合并的PR和模板链接https://github.com/PaddlePaddle/docs/blob/develop/docs/guides/model_convert/convert_from_pytorch/api_difference/pytorch_api_mapping_format_cn.md
```python
import paddle
x = paddle.to_tensor([[10, 30, 20], [60, 40, 50]])
idx = paddle.to_tensor([[0]])
y = x.take_along_axis(idx, 0)
```

Comment (zpc2024) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.take_along_dim.md]: 感谢评论，已经修改了。

File: docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+|out         | -         |  表示输出的 Tensor ， Paddle 无此参数，需要进行转写。 |
+
+### 转写示例
+#### out：指定输出
+```python
+# Pytorch 写法
+torch.linalg.matrix_power(x, 3， out = y)
+
+# Paddle 写法
+y = paddle.linalg.matrix_power(x, 3)
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md]: 抱歉，昨天是我没看到最新的标准，这种指定输出的还是写为 `paddle.assign(paddle.linalg.matrix_power(x, 3), y)` 吧，抱歉~

Comment (zpc2024) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/linalg/torch.linalg.matrix_power.md]: 好的，修改好了~

**Review by Fancy-hjyp — APPROVED** [review]
LGTM

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.dsplit.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+        axis=0,
+        name=None)
+```
+
+Paddle 相比 PyTorch 支持更多其他参数，具体如下：
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| input          |  x           | 输入多维 Tensor ，仅参数名不一致。  |
+| indices_or_sections         | num_or_sections         | int 或者仅含有 int 的 list 或者 tuple ，用于分割的参数，仅参数名不一致。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.dsplit.md]: 这个参数是不是和torch.split一样，属于 参数不一致

https://github.com/PaddlePaddle/docs/blob/develop/docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.split.md

Comment (zpc2024) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.dsplit.md]: 感谢评论，您提到的文档版本是torch2.0版本的，我查阅1.13的文档
https://pytorch.org/docs/1.13/generated/torch.dsplit.html#torch.dsplit
https://pytorch.org/docs/1.13/generated/torch.tensor_split.html#torch-tensor-split
确认了indices_or_sections 参数和 num_or_sections 参数，仅参数名不一致。
paddle.split axis参数应该采用2，而不是默认0，已修改。

Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.dsplit.md]: 了解了。`torch.dsplit`和`torch.split`确实不一样，前者指定块的个数，后者指定块的大小

**Review by zhwesky2010 — APPROVED** [review]

**Comment by luotao1** [comment]
hi, @zpc2024 
- 非常感谢你对飞桨框架的贡献，我们正在运营一个PFCC组织，会通过定期分享技术知识与发布开发者主导任务的形式持续为飞桨框架做贡献，详情可见 https://github.com/luotao1 主页说明。
- 如果你对PFCC有兴趣，请发送邮件至 [ext_paddle_oss@baidu.com](mailto:ext_paddle_oss@baidu.com)，我们会邀请你加入~
